packages/modeling/src/operations/offsets/offsetPath2.js

```diff
@@ -74,17 +74,17 @@ export const offsetPath2 = (options, geometry) => {
 
   const closed = geometry.isClosed
   const points = path2.toPoints(geometry)
-  if (points.length === 0) throw new Error('the given geometry cannot be empty')
+  if (points.length === 0) return geometry
```

Comment:
Strongly disagree!

The point of immutable data structures, is that we can return the same empty geometry and be confident that it's okay. If we start making copies of object unnecessarily, it will cost a lot in memory usage and efficiency.

Reply:
Well then, I think that the throw is a better solution.

Reply:
No, throw is much worse in my opinion! If you really want me to return a copy I will do that, I really do **not** want to throw.

The geometry is not "incorrect". The empty geometry is a perfectly valid geometry. We should absolutely allow users to `expand` or `offset` or `project` any set of valid geometries, some of which might be empty. If JSCAD throws an exception, you can't perform these operations without painstakingly checking every geometry if it's empty or not.

What does it mean to expand an empty geometry? Empty geometry.
What does it mean to offset an empty geometry? Empty geometry.
What does it mean to extrude an empty geometry? Empty geometry.

Please do not make me throw an error here.

And if not throwing, what is the advantage of making a copy instead of returning the original?

Reply:
I'm not totally disagreeing. I agree that returning a geometry is better choice than throwing an error, but I'm not convinced that returning the original is correct.

There are lots of operations that will accept empty geometries but those operations don't return the original. Consistency is important.

I cannot come up an immediate use case, but if a geometry has a special attribute set then the result should be something else.

I acknowledge that a new object creates an allocation. But users know these are heavy operations. The use of create() is as lightweight as possible.

Reply:
Alternatively, just let the extrude happen on an empty set of points.